docs/spec/mint/begin_block.md

```diff
@@ -26,3 +26,26 @@ NextInflation(params Params, bondedRatio sdk.Dec) (inflation sdk.Dec) {
 	}
 
 	return inflation
+```
+
+## NextAnnualProvisions
+
+Calculate the annual provisions based on current total supply and inflation
+rate. This parameter is calculated once per block. 
+
+```
+NextAnnualProvisions(params Params, totalSupply sdk.Dec) (provisions sdk.Int) {
+	provisionsDec = Inflation * totalSupply
+	return provisionsDec.Truncate()
```

Comment:
Why do we need to truncate here - can we instead truncate after calculating the per-block provisions? That would be more accurate.

Reply:
I mean we can but then we're doing a decimal calculation every block which is considerably less efficient than an Int operation from what I've seen... I think we should keep it as is

Reply:
okay - I agree a benchmark is due here for this PR

Reply:
benchmarked 200 ns vs. 429 ns. - it's a non-freaking issue - switched to decimals (thanks val)

Reply:
Thanks - looks like we still need to update the spec though.